lib/ex_azure_key_vault.ex

```diff
@@ -55,6 +56,108 @@ defmodule ExAzureKeyVault.Client do
     }
   end
 
+  @doc """
+  Connects to Azure Key Vault (Using MSI - Managed Service Identity).
+
+  ## Examples
+
+  When defining environment variables and/or adding to configuration.
+
+      $ export AZURE_VAULT_NAME="my-vault"
+
+      # Config.exs
+      config :ex_azure_key_vault,
+        azure_vault_name: {:system, "AZURE_VAULT_NAME"}
+
+      iex(1)> ExAzureKeyVault.Client.msi_connect()
+      %ExAzureKeyVault.Client{
+        api_version: "2016-10-01",
+        bearer_token: "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...",
+        vault_name: "my-vault"
+      }
+
+  Passing custom parameters.
+
+      iex(1)> ExAzureKeyVault.Client.msi_connect("custom-vault")
+      %ExAzureKeyVault.Client{
+        api_version: "2016-10-01",
+        bearer_token: "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...",
+        vault_name: "custom-vault"
+      }
+
+  """
+  @spec smart_connect() :: Client.t() | {:error, any}
+  @spec smart_connect(String.t() | nil) :: Client.t() | {:error, any}
+  def smart_connect(vault_name \\ nil) do
+    with {:error, error1} <- msi_connect!(vault_name),
+         {:error, error2} <- connect!(vault_name),
+         {:error, error3} <- cert_connect!(vault_name) do
+      {:error, "#{error1},#{error2},#{error3}"}
+    else
+      client -> client
+    end
+  end
+
+  @doc """
+  Connects to Azure Key Vault (Using MSI - Managed Service Identity).
+
+  ## Examples
+
+  When defining environment variables and/or adding to configuration.
+
+      $ export AZURE_VAULT_NAME="my-vault"
+
+      # Config.exs
+      config :ex_azure_key_vault,
+        azure_vault_name: {:system, "AZURE_VAULT_NAME"}
+
+      iex(1)> ExAzureKeyVault.Client.msi_connect()
+      %ExAzureKeyVault.Client{
+        api_version: "2016-10-01",
+        bearer_token: "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...",
+        vault_name: "my-vault"
+      }
+
+  Passing custom parameters.
+
+      iex(1)> ExAzureKeyVault.Client.msi_connect("custom-vault")
+      %ExAzureKeyVault.Client{
+        api_version: "2016-10-01",
+        bearer_token: "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...",
+        vault_name: "custom-vault"
+      }
+
+  """
+  @spec msi_connect!() :: Client.t() | {:error, any}
+  @spec msi_connect!(String.t() | nil) ::
+          Client.t() | {:error, any}
+  def msi_connect!(vault_name \\ nil) do
+    try do
+      msi_connect(vault_name)
+    rescue
+      e in ArgumentError -> {:error, e}
+    end
+  end
+
+  @spec msi_connect() :: Client.t() | {:error, any}
+  @spec msi_connect(String.t() | nil) ::
+          Client.t() | {:error, any}
+  def msi_connect(vault_name \\ nil) do
+    vault_name = get_env(:azure_vault_name, vault_name)
+    endpoint = System.get_env("IDENTITY_ENDPOINT")
```

Comment:
I'd suggest you move the `IDENTITY_ENDPOINT` and `IDENTITY_HEADER` env variables to the config files for each environment. This way you can leverage the `get_env()` function for them as well.